ext/rugged/rugged_commit.c

```diff
@@ -758,12 +758,43 @@ static VALUE rb_git_commit_create_to_s(VALUE self, VALUE rb_repo, VALUE rb_data)
 	return ret;
 }
 
+/*
+ *    Rugged::Commit.create_with_signature(repo, content, signature, field_name = "gpgsig") -> oid
+ *
+ *  Create a commit from the +content+ string and the +signature+,
+ *  adding this data to the +field_name+ header field in the resulting
+ *  commit.
+ *
+ *  Returns the new commit's object id.
+ *
+ */
+static VALUE rb_git_commit_create_with_signature(int argc, VALUE *argv, VALUE self)
+{
+	int error;
+	git_oid id;
+	const char *field;
+	git_repository *repo;
+	VALUE rb_repo, rb_content, rb_signature, rb_field = Qnil;
+
+	rb_scan_args(argc, argv, "31", &rb_repo, &rb_content, &rb_signature, &rb_field);
+
+	rugged_check_repo(rb_repo);
+	Data_Get_Struct(rb_repo, git_repository, repo);
+
+	field = rb_field == Qnil ? NULL : StringValueCStr(rb_field);
```

Comment:
You might want to add `CHECK_TYPE` calls for `rb_field`, `rb_content` and `rb_signature`.